packages/app/src/systems/Vault/services/VaultServer.ts

```diff
@@ -160,11 +177,35 @@ export class VaultServer extends EventEmitter {
     vaultId,
     password,
   }: VaultInputs['exportVault']): Promise<string> {
-    await this.manager.unlock(password);
+    if (password) await this.manager.unlock(password);
     const vault = await this.manager.exportVault(vaultId);
     return vault.secret || '';
   }
 
+  async getPositionsToConfirm({
+    words,
+    limit = 9,
+  }: VaultInputs['getPositionsToConfirm']): Promise<
+    VaultOutputs['getPositionsToValidate']
+  > {
+    const randomWords = shuffle(words).slice(0, limit);
+    const positions = shuffle(
+      randomWords.map((word) => words.indexOf(word) + 1)
+    ).sort((a, b) => a - b);
+
+    return { positions };
+  }
+
+  async confirmMnemonic({
+    words,
+  }: VaultInputs['confirmMnemonic']): Promise<boolean> {
+    const mnemonic = (await this.exportVault({ vaultId: 0 })).split(' ');
+    const isValid = words.every((word, index) => {
+      return word === mnemonic[index];
+    });
+    return isValid;
```

Comment:
I think here you can create a HDWallet using the words provided and do the match by first address this will remove the need to have the vault unlocked.